src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -112,27 +112,27 @@ private void ResumeRestartIfScheduled()
                 using (System.Threading.ExecutionContext.SuppressFlow())
                 {
                     _typedLogger.LogDebug("Resuming scheduled restart.");
-                    Task.Run(async () => await ScheduleRestartAsync());
+                    Task.Run(async () => await ScheduleRestartAsync("Running restart requested while in a restart suspension scope."));
                 }
             }
         }
 
-        private async Task ScheduleRestartAsync(bool shutdown)
+        private async Task ScheduleRestartAsync(string reason, bool shutdown)
         {
             _restartScheduled = true;
             if (shutdown)
             {
                 _shutdownScheduled = true;
             }
 
-            await ScheduleRestartAsync();
+            await ScheduleRestartAsync(reason);
         }
 
-        private async Task ScheduleRestartAsync()
+        private async Task ScheduleRestartAsync(string reason)
         {
             if (Interlocked.Read(ref _suspensionRequestsCount) > 0)
             {
-                _logger.LogDebug("Restart requested while currently suspended. Ignoring request.");
+                _logger.LogDebug($"Restart requested due to '{reason}' while currently suspended. Ignoring request.");
```

Comment:
Recommend rewording to: "Restart requested while currently suspended. Reason: '{reason}'. Ignoring request."

Reply:
Updated message.